tests/mbt/model/ccv.qnt

```diff
@@ -654,6 +840,168 @@ module ccv {
         }
     }
 
+    // Makes the given consumer slash the given validator for an infraction that happened while
+    // the provided vscId was in effect.
+    // Note that the model does not model the infractions themselves yet,
+    // e.g. we do not model validators actually double-signing or going down.
+    pure def sendSlashRequest(
+        currentState: ProtocolState,
+        consumer: Chain,
+        consumerNode: Node,
+        vscId: VscId,
+        downtime: bool): Result = {
+        val consumerState = currentState.consumerStates.get(consumer)
+        if (not(isRunningConsumer(consumer, currentState.providerState))) {
+            Err("Consumer is not currently a consumer - must have 'running' status!")
+        } else if (downtime and consumerState.outstandingDowntime.contains(consumerNode)) {
+            Err("Consumer already has outstanding downtime for validator!")
+        } else {
+            val vscPacket = consumerState.receivedVscPackets.select(packet => packet.id == vscId)[0]
+            val packet = Slash({
+                validator: consumerNode,
+                id: vscId,
+                downtime: downtime,
+                sendingTime: consumerState.chainState.runningTimestamp,
+                timeoutTime: consumerState.chainState.runningTimestamp + CcvTimeout.get(consumer),
+                valPower: vscPacket.validatorSet.get(consumerNode)
+
+            })
+            val newConsumerState = 
+                commitPacketOnConsumer(consumerState, packet)
+                // add the consumer address to the list of outstanding slashes
+                .with("outstandingDowntime", consumerState.outstandingDowntime.union(Set(consumerNode)))
+            Ok(currentState.with("consumerStates", currentState.consumerStates.set(consumer, newConsumerState)))
+        }
+    }
+
+    // receives a given packet on the provider. The arguments are the consumer chain that sent the packet, and the packet itself.
+    pure def recvPacketOnProvider(currentState: ProtocolState, sender: Chain, packet: Packet): Result = {
+        match packet {
+            | VscMatured(p) => recvVscMaturedPacketOnProvider(currentState, sender, p)
+            | Slash(p) => recvSlashPacketOnProvider(currentState, sender, p)
+            | _ => Err("Do not know how to handle this packet type")
+        }
+    }
+
+    pure def recvSlashPacketOnProvider(currentState: ProtocolState, sender: Chain, packet: SlashPacket): Result = {
+        // determine the jail duration for the validator:
+        // if the packet is a downtime slash, the jail duration is DowntimeJailTime
+        // otherwise, the validator is jailed until the DoubleSignJailTime
+        val jailEndTime = if (packet.downtime) {
+            packet.sendingTime + DowntimeJailTime
+        } else {
+            packet.sendingTime + DoubleSignJailEndTime
```

Comment:
The provider is only handling `SlashPackets` for downtime infractions. Packets with double-signing infractions are dropped on receipt.
See https://github.com/cosmos/interchain-security/blob/main/x/ccv/provider/keeper/relay.go#L313-L329.

Reply:
Oh yea, I missed it. Yes please, it isn't clear that `DoubleSignJailEndTime` and `DoubleSignSlashPercentage` are never used to update the provider states.